uaclient/api/u/pro/services/list/v1.py

```diff
@@ -0,0 +1,206 @@
+    resources = machine_token_file.machine_token.get("availableResources")
+    if not resources:
+        resources = get_available_resources(cfg)
+    inapplicable_resources = {
+        resource["name"]: resource.get("description")
+        for resource in sorted(resources, key=lambda x: x.get("name", ""))
+        if not resource.get("available")
+    }
+    for resource in resources:
+        try:
+            ent = entitlement_factory(cfg=cfg, name=resource.get("name", ""))
+        except exceptions.EntitlementNotFoundError:
+            LOG.debug(
+                "Ignoring availability of unknown service %s from contract "
+                "server",
+                resource.get("name", "without a 'name' key"),
+            )
+            continue
+
+        services_list.append(
+            _get_service_info(cfg, ent, inapplicable_resources)
+        )
+    services_list.sort(key=lambda x: x.name)
+    return ServiceListResult(services=services_list)
+
+
+def unattached_services(cfg: UAConfig) -> ServiceListResult:
+    services_list = []
+    resources = get_available_resources(cfg)
+    for resource in resources:
+        available = True if resource.get("available") else False
+        try:
+            ent = entitlement_factory(cfg=cfg, name=resource.get("name", ""))
+        except exceptions.EntitlementNotFoundError:
+            LOG.debug(
+                "Ignoring availability of unknown service %s from contract "
+                "server",
+                resource.get("name", "without a 'name' key"),
+            )
+            continue
+        service = ServiceInfo(
+            name=resource.get("presentedAs", resource["name"]),
+            desc=ent.description,
+            available=available,
+            entitled=None,
+        )
+        services_list.append(service)
+    services_list.sort(key=lambda x: x.name)
+    return ServiceListResult(services=services_list)
+
+
+def _get_service_info(
+    cfg: UAConfig, ent: UAEntitlement, inapplicable_resources: Dict
+) -> ServiceInfo:
+    contract_status = ent.contract_status()
+    entitled = None
+    if contract_status == ContractStatus.UNENTITLED:
+        entitled = False
+    elif contract_status == ContractStatus.ENTITLED:
+        entitled = True
+    available = True if ent.name not in inapplicable_resources else False
+
+    service_info = ServiceInfo(
+        name=ent.presentation_name,
+        desc=ent.description,
+        available=available,
+        entitled=entitled,
+    )
+    return service_info
+
+
+endpoint = APIEndpoint(
+    version="v1",
+    name="ServicesList",
+    fn=_get_service_list,
+    options_cls=None,
+)
+
+_doc = {
+    "introduced_in": "35",
+    "requires_network": False,
+    "example_python": """
+from uaclient.api.u.pro.services.list.v1 import get_service_list
+
+result = get_service_list()
+""",  # noqa: E501
+    "result_class": ServiceListResult,
+    "exceptions": [],
```

Comment:
At least  `ContractAPIError` should be here since we talk to that. Maybe look for others as well?